src/patterns/gen-ai/aws-text-to-sql/index.ts

```diff
@@ -0,0 +1,1219 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+import * as path from 'path';
+import * as cdk from 'aws-cdk-lib';
+import { Aws, Duration, aws_events_targets as eventsTarget } from 'aws-cdk-lib';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import * as events from 'aws-cdk-lib/aws-events';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as lambda from 'aws-cdk-lib/aws-lambda';
+import * as logs from 'aws-cdk-lib/aws-logs';
+import * as rds from 'aws-cdk-lib/aws-rds';
+import * as s3 from 'aws-cdk-lib/aws-s3';
+import * as secretsmanager from 'aws-cdk-lib/aws-secretsmanager';
+import * as sqs from 'aws-cdk-lib/aws-sqs';
+import * as stepfunctions from 'aws-cdk-lib/aws-stepfunctions';
+import * as tasks from 'aws-cdk-lib/aws-stepfunctions-tasks';
+import { NagSuppressions } from 'cdk-nag';
+import { Construct } from 'constructs';
+import {
+  BaseClass,
+  BaseClassProps,
+} from '../../../common/base-class/base-class';
+import { ConstructName } from '../../../common/base-class/construct-name-enum';
+import { buildCustomResourceProvider } from '../../../common/helpers/custom-resource-provider-helper';
+import * as eventBridgeHelper from '../../../common/helpers/eventbridge-helper';
+
+import { buildDockerLambdaFunction } from '../../../common/helpers/lambda-builder-helper';
+import * as s3BucketHelper from '../../../common/helpers/s3-bucket-helper';
+import {
+  generatePhysicalNameV2,
+  lambdaMemorySizeLimiter,
+} from '../../../common/helpers/utils';
+import * as vpcHelper from '../../../common/helpers/vpc-helper';
+import { DockerLambdaCustomProps } from '../../../common/props/DockerLambdaCustomProps';
+import { ServiceEndpointTypeEnum } from '../../../patterns/gen-ai/aws-rag-appsync-stepfn-kendra/types';
+
+export enum DatabaseType {
+  AURORA = 'Aurora',
+  RDS = 'RDS',
+}
+
+export enum DbName {
+  MYSQL = 'MySQL',
+  POSTGRESQL = 'PostgreSQL',
```

Comment:
Seems the option for POSTGRESQL is commented further down in the file (createX methods):

```
// case DbName.POSTGRESQL:
      //   // Add code for RDS PostgreSQL instance
      //   return undefined;
```

Should we remove that option ?

Reply:
Postgresql is being worked under another PR.

Reply:
I think we can remove it then for now, while it is implemented to prevent any confusion

Reply:
This is already removed with a91c03fd228eea36e3d0070f4e97d92ba2585903